src/compiler/checker.ts

```diff
@@ -22688,6 +22827,16 @@ namespace ts {
             return undefined;
         }
 
+        function isLiteralDynamicName(name: ComputedPropertyName) {
+            name = getParseTreeNode(name, isComputedPropertyName);
+            if (name) {
+                // TODO(rbuckton): ESSymbolLiteral
+                const nameType = checkComputedPropertyName(name);
+                return (nameType.flags & TypeFlags.StringOrNumberLiteral) !== 0;

```

Comment:
I'm not sure it has a major impact on performance, but `!!` requires coercion from SMI to boolean, while `!== 0` does not.